salt/modules/yumpkg.py

```diff
@@ -1518,13 +1518,17 @@ def install(name=None,
             norm_epoch = lambda x, y: x.split(':', 1)[-1] \
                 if ':' not in y \
                 else x
+
             cver = old_as_list.get(pkgname, [])
             if reinstall and cver:
                 for ver in cver:
-                    ver = norm_epoch(ver, version_num)
+                    # if diff_attr was provided, version number is directly
+                    # in ver, otherwise its a dict.
+                    ver2 = ver['version'] if diff_attr else ver
+                    ver2 = norm_epoch(ver2, version_num)
```

Comment:
This will not handle epochs properly. In fact, I dislike the entire way `diff_attr` was implemented, now that I think about it, since it destroys the information we get from pkg.list_pkgs. Please hold off on this, I have changes to make to how `diff_attr` works in the oxygen branch, which may require that this PR be updated or even unnecessary.